File: android/app/src/main/java/com/teotuyetvoi/drivereader/HardwareKeyModule.java
package com.teotuyetvoi.drivereader;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.*;
import com.facebook.react.modules.core.DeviceEventManagerModule;

public class HardwareKeyModule extends ReactContextBaseJavaModule {
    private static ReactApplicationContext reactContext;

    public HardwareKeyModule(ReactApplicationContext context) {
        super(context);
        reactContext = context;
    }

    @NonNull
    @Override
    public String getName() {
        return "HardwareKeyModule";
    }

    public static void sendKeyEvent(int keyCode) {
        if (reactContext == null) return;

        WritableMap map = Arguments.createMap();
        map.putInt("keyCode", keyCode);

        reactContext
            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
            .emit("HardwareKeyPress", map);
    }

    @ReactMethod
    public void addListener(String eventName) {}

    @ReactMethod
    public void removeListeners(Integer count) {}
}

File: android/app/src/main/java/com/teotuyetvoi/drivereader/HardwareKeyModule.kt
package com.teotuyetvoi.drivereader

import com.facebook.react.bridge.*
import com.facebook.react.modules.core.DeviceEventManagerModule

class HardwareKeyModule(private val reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String = "HardwareKeyModule"

    companion object {
        private var instance: HardwareKeyModule? = null

        fun setInstance(module: HardwareKeyModule) {
            instance = module
        }

        fun sendKeyEvent(keyCode: Int) {
            instance?.reactContext
                ?.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
                ?.emit("HardwareKeyPress", Arguments.createMap().apply {
                    putInt("keyCode", keyCode)
                })
        }
    }

    init {
        setInstance(this)
    }

    @ReactMethod
    fun addListener(eventName: String) {}

    @ReactMethod
    fun removeListeners(count: Int) {}
}